### DISCUSSION TITLE
Provide documentation about CSI Driver with harvesterhci.io

### DISCUSSION THREAD

**Discussion by vanvuongngo** [siderolabs/talos#9823] [Ideas]
Hi,

congrats for the great talos project. I really like the awesome security concept and low maintenance efforts.

As a starter I would like to use talos with https://harvesterhci.io/ which is a hyperconverged infrastructure solution.
It provides network balancers, persistent volumes claims with longhorn and a lot more.
These a things I am missing when using talos.

I got this documentation from the harvesterhci team about the CSI driver which can be use to claim persistent volumes (longhorn). But I can not get it work: 
https://docs.harvesterhci.io/v1.3/rancher/csi-driver

Do you have a clue how to proceed?

best regards

**Comment by smira** [comment]
From the looks of it, Harvester has deep top-down integration and it comes with its own operating system. It doesn't seem to be something you can "install" on top of something else rather than what they provide.

**Reply by vanvuongngo** [reply]
Hello @smira,

thanks for your quick reply.
Yes, harvester is like proxmox and has its own operating system.
I have created VMs for my Talos cluster with harvester. And for the k8s deployments I want to claim persistent volumes with harvester.
Rancher has harvester integration out of the box as both are from Suse. I can also create k3s clusters with Rancher.
But I want to use Talos instead of a special cluster for Rancher and with that you can only create k3s clusters.

So Talos + Harvester (LB, PVC) would be a complete dream team for the Edge IMO ... Easy to set up, maintain and include everything a k8s application needs.

I'd be happy if that somehow worked.

**Comment by vanvuongngo** [comment]
I also placed the idea on the harvester project: https://github.com/harvester/harvester/issues/6834

@w13915984028 tried to help: "You may refer https://docs.harvesterhci.io/v1.3/rancher/csi-driver. When your guest cluster is running workload which claims for persistant volume, then the guest cluster needs kind of CSI driver to support it."
